### PR TITLE
libuv: modernize

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

Actually the main goal is to generate gcc11 binaries for https://github.com/conan-io/conan-center-index/pull/8746

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by Croydon — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

File: recipes/libuv/all/conanfile.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ def package(self):
         cmake.install()
 
     def package_info(self):
-        self.cpp_info.names["pkg_config"] = "libuv"
+        self.cpp_info.set_property("pkg_config_name", "libuv")
```
Comment (uilianries) [inline recipes/libuv/all/conanfile.py:75]:
```suggestion
        self.cpp_info.set_property("pkg_config_name", "libuv")
        self.cpp_info.set_property("cmake_file_name", "LibUV")
        self.cpp_info.set_property("cmake_target_name", "LibUV::LibUV")
        # TODO: Remove after Conan 2.0
        self.cpp_info.name["cmake_find_package"] = "LibUV"
        self.cpp_info.name["cmake_find_package_multi"] = "LibUV"
```
Cmake official modules have libuv listed as LibUV: https://gitlab.kitware.com/cmake/cmake/-/blob/v3.22.1/Source/Modules/FindLibUV.cmake

File: recipes/libuv/all/test_package/conanfile.py [hunk 2]
```diff
@@ -2,8 +2,8 @@
 import os
 
 
-class libuvTestConan(ConanFile):
-    settings = "os", "compiler", "build_type", "arch"
+class TestPackageConan(ConanFile):
+    settings = "os", "arch", "compiler", "build_type"
     generators = "cmake"
```
Comment (uilianries) [inline recipes/libuv/all/test_package/conanfile.py:7]:
```suggestion
    generators = "cmake", "cmake_find_package_multi"
```

File: recipes/libuv/all/test_package/CMakeLists.txt [hunk 3]
```diff
@@ -4,6 +4,5 @@ project(test_package C)
 include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
 conan_basic_setup()
```
Comment (uilianries) [inline recipes/libuv/all/test_package/CMakeLists.txt:5]:
```suggestion
conan_basic_setup(TARGETS)

find_package(LibUV CONFIG REQUIRED)

```

File: recipes/libuv/all/test_package/CMakeLists.txt [hunk 4]
```diff
@@ -4,6 +4,5 @@ project(test_package C)
...
 
-add_executable(${PROJECT_NAME} example.c)
-
-conan_target_link_libraries(${PROJECT_NAME})
+add_executable(${PROJECT_NAME} test_package.c)
+target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
```
Comment (uilianries) [inline recipes/libuv/all/test_package/CMakeLists.txt:8]:
```suggestion
target_link_libraries(${PROJECT_NAME} LibUV::LibUV)
```

**Review by uilianries — COMMENTED** [review]
CMake has an official module for libuv.

File: recipes/libuv/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/libuv/all/conanfile.py:75]: If it's not listed in https://cmake.org/cmake/help/latest/manual/cmake-modules.7.html, it's not official. The FindLibUV you are showing is a internal build detail of CMake.

**Review by uilianries — APPROVED** [review]

**Pull request merged**